### ISSUE TITLE
Unable to use JMock 2.6.0 with Java 8

### ISSUE THREAD

**Issue by muminc** [jmock-developers/jmock-library#60]
Out of curiosity, tried to build an existing project (which uses 7) with Java 8 (source = 1.8, target=1.8) but I see many of these errors  emitted by the  Java 8 Compiler.

```
error: reference to with is ambiguous  both method with(Matcher<Float>) in Expectations and method with(Matcher<Double>) in Expectations match
````

**Comment by bsiver** [comment]
Thanks @mread, I had some issues using parameterized types in expectations after upgrading to JDK 8:
```java
oneOf(something).method(with(any(X.class)))
```
Started to generate a compiler error:
`The method something(X<Y>) in the type Something is not applicable for the arguments (X).`

Changing the expectation to
```java
oneOf(something).method(with.is(any(X.class)))
```
fixed the compiler error.

**Comment by olibye** [comment]
Solved in 2.8.0, in maven central now.
